src/Workspaces/VisualBasic/Portable/Extensions/ExpressionSyntaxExtensions.vb

```diff
@@ -1476,7 +1476,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             If name.IsKind(SyntaxKind.GenericName) Then
                 If (name.IsParentKind(SyntaxKind.CrefReference)) OrElse ' cref="Nullable(Of T)"
                    (name.IsParentKind(SyntaxKind.QualifiedName) AndAlso name.Parent?.IsParentKind(SyntaxKind.CrefReference)) OrElse ' cref="System.Nullable(Of T)"
-                   (name.IsParentKind(SyntaxKind.QualifiedName) AndAlso name.Parent?.IsParentKind(SyntaxKind.QualifiedName) AndAlso name.Parent?.Parent?.IsParentKind(SyntaxKind.CrefReference)) Then  ' cref="System.Nullable(Of T).Value"
+                   (name.IsParentKind(SyntaxKind.QualifiedName) AndAlso (name.Parent?.IsParentKind(SyntaxKind.QualifiedName)).GetValueOrDefault() AndAlso name.Parent.Parent?.IsParentKind(SyntaxKind.CrefReference)) Then  ' cref="System.Nullable(Of T).Value"
```

Comment:
> (name.Parent?.IsParentKind(SyntaxKind.QualifiedName)).GetValueOrDefault() [](start = 71, length = 74)

This is just an optimization.

Reply:
I'm confused: is this ensuring the same optimization happens once Roslyn is built with a newer version of itself after this PR is merged? We absolutely don't care about that in the IDE cases, so we should just keep the code simple.

Reply:
> is this ensuring the same optimization happens once Roslyn is built with a newer version of itself after this PR is merged?

No. This code was not optimized by the compiler before and isn't optimized now. Addition of GetValueOrDefault() call eliminates the need for "Nullable" logic. If you insist I can revert this change. 

---
In reply to: [327884076](https://github.com/dotnet/roslyn/pull/38802#discussion_r327884076) [](ancestors = 327884076)

Reply:
Ah, I misunderstood -- didn't realize the null case would continue to evaluate.